scripts/db_migrator.py

```diff
@@ -210,6 +366,28 @@ def version_1_0_4(self):
         """
         log.log_info('Handling version_1_0_4')
 
+        # Check ASIC type, if Mellanox platform then need DB migration
+        if self.asic_type == "mellanox":
+            speed_list = ['1000', '10000', '25000', '40000', '50000', '100000', '200000', '400000']
+            cable_len_list = ['5m', '40m', '300m']
+            buffer_pools = self.configDB.get_table('BUFFER_POOL')
+            buffer_profiles = self.configDB.get_table('BUFFER_PROFILE')
+            buffer_pgs = self.configDB.get_table('BUFFER_PG')
+            if self.mellanox_buffer_migrator.mlnx_migrate_buffer_pool_size('version_1_0_4', 'version_1_0_5') \
+               and self.mellanox_buffer_migrator.mlnx_migrate_buffer_profile('version_1_0_4', 'version_1_0_5') \
+               and self.migrate_config_db_buffer_tables_for_dynamic_calculation(speed_list, cable_len_list, '0') \
+               and self.prepare_dynamic_buffer_for_warm_reboot(buffer_pools, buffer_profiles, buffer_pgs):
```

Comment:
if we need to run backward compatible mode on mellanox, then we should skip dynamic calc migration. Also how do we know what is our current mode?

Reply:
The current mode can be fetched from `CONFIG_DB.DEVICE_METADATA|localhost`